application/classes/Ushahidi/Repository/Dataprovider.php

```diff
@@ -28,6 +28,25 @@ class Ushahidi_Repository_Dataprovider implements
 	// ReadRepository
 	public function getEntity(Array $data = null)
 	{
+		// translating labels and description for data-providers
+		if($data !== null && $data['options']) {
+			foreach ($data['options'] as $option_key => $option) {
+				foreach($option as $value_key => $value) {
```

Comment:
Couldn't you drop this foreach and just do
```
$options['label']['label'] = __($options['label']['label']);
$options['description']['label'] = __($options['description']['label']);
```

with some `isset` and `gettype` checks in there too. Looping to find known keys just makes this messier.

Reply:
Neat! Fixed now!